src/lib/mongoBackend/mongoUpdateSubscription.cpp

```diff
@@ -803,15 +802,16 @@ void updateInCache
 
   if (mscInsert == 0)  // 0: Insertion was really made
   {
-    subCacheUpdateStatisticsIncrement();
-
-    if (subCacheP != NULL)
+    subCacheStatisticsIncrementUpdates();
+  // FIXME PR: I understand this block is not needed with std::map based
+  // logic
+  /*  if (subCacheP != NULL)
     {
       LM_T(LmtSubCache, ("Calling subCacheItemRemove"));
       subCacheItemRemove(subCacheP);
     }
+   */
```

Comment:
Yeah, looking closer at it, it becomes clear. The cache now works differently and the remove should never be done. We no longer "add then remove" items but they are overwritten